### PR TITLE
Enable COWArrayOpts on OSSA

### PR DESCRIPTION
**Pull request by meg-gupta**
Additional handling of copy_value/destroy_value/load[copy]/
begin_borrow/end_borrow is needed to support OSSA.

TODO: Support handling of 2d array in the pass for OSSA.
Currently hoisting of loads and borrows are not supported in OSSA

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci benchmark

File: lib/SILOptimizer/LoopTransforms/ArrayOpt.h [hunk 1]
```diff
@@ -184,6 +187,7 @@ class StructUseCollector {
         // Found a use of the struct at the given access path.
         if (auto *LoadI = dyn_cast<LoadInst>(UseInst)) {
           StructLoads.push_back(LoadI);
+          StructAddressUsers.push_back(LoadI);
```
Comment (atrick) [inline lib/SILOptimizer/LoopTransforms/ArrayOpt.h:190]: Please update the file comment
```
/// StructLoads records loads of the struct value.
/// StructAddressUsers records other uses of the struct address.
```

**Review by atrick — APPROVED** [review]
LGTM. I just want to point out that the TODO's are actually regressions so they need to be tracked somehow.

File: lib/SILOptimizer/LoopTransforms/ArrayOpt.h (hunk 1 above)
Comment (meg-gupta) [reply lib/SILOptimizer/LoopTransforms/ArrayOpt.h:190]: I'll update this in a follow up commit.

**Pull request merged**